### PR TITLE
Feature: Field validation, internal refactoring 1.2.0

### PR DESCRIPTION
**Pull request by f3rno**
### Description:
Adds support for built-in model field validation, and refactors core `Model` method signatures slightly decreasing the number of parameters that must be provided for model definitions. Also adds more tests!

### TODO
- [x] add validators to all models
- [x] add more (as complete as possible) unit/integration tests

### Breaking changes:
- [x] `Model` constructor now takes a map instead of a series of arguments
- [x] `Model.unserialize` now takes a map instead of a series of arguments
- [x] `Order._modifyFlag()` is now public as `Order.modifyFlag()`

### New features:
- [x] `fieldKeys` no longer needs to be provided when creating a new model
- [x] `boolFields` is now optional when creating a new model
- [x] added `Model.validate()`
- [x] added validation to all models
- [x] added `isCollection`
- [x] switched to `async/await` where possible
- [x] added `description` and `meta` field support to `Wallet` model
- [x] added `nyc` and set/enforced limits
- [x] added husky `npm test` pre-commit hook
- [x] more tests!

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [ ] Tests added or updated
- [ ] Documentation updated

### CODE REVIEW

**Comment by JacobPlaster** [comment]
@f3rno how do you want to handle the breaking changes? Do we need to update the bitfinex-api-node lib?

File: lib/funding_credit.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
 'use strict'
 
+const numberValidator = require('./validators/number')
+const amountValidator = require('./validators/amount')
+const symbolValidator = require('./validators/symbol')
```
Comment (JacobPlaster) [inline lib/funding_credit.js]: Nice, I really like the separation of the validator functins

Comment (f3rno) [reply lib/funding_credit.js]: Thanks :slightly_smiling_face:

**Comment by f3rno** [comment]
@JacobPlaster there should be no breaking changes with the old API, only a private method was made public, and the base `Model` class was modified (which isn't being used anywhere externally). Other API changes are additions.

**Pull request merged**